File: Others/easy/GenerateDocument.py
def generateDocument(characters, document):
    count_chars = {}
    for char in characters:
        count_chars[char] = count_chars.get(char, 0) + 1

    count_doc = {}
    for char in document:
        count_doc[char] = count_doc.get(char, 0) + 1

    for char in count_doc:
        if char not in count_chars or count_doc[char] > count_chars[char]:
            return False
    return True

File: Golang/easy/generate-document.go
package main

func makeFreq(characters string) map[rune]int {
	res := make(map[rune]int)
	runes := []rune(characters)
	for _, char := range runes {
		res[char] += 1
	}

	return res
}

func GenerateDocument(characters string, document string) bool {
	charsFreq := makeFreq(characters)
	docFreq := makeFreq(document)

	for char, occ := range docFreq {
		val, ok := charsFreq[char]
		if !ok || occ > val {
			return false
		}
	}
	return true
}
